src/FsAutoComplete/CodeFixes/IgnoreExpression.fs

```diff
@@ -0,0 +1,96 @@
+module FsAutoComplete.CodeFix.IgnoreExpression
+
+open FSharp.Compiler.Syntax
+open FSharp.Compiler.SyntaxTrivia
+open FSharp.Compiler.Text
+open FSharp.UMX
+open FsToolkit.ErrorHandling
+open Ionide.LanguageServerProtocol.Types
+open FsAutoComplete.CodeFix.Types
+open FsAutoComplete
+open FsAutoComplete.LspHelpers
+
+let title = "Ignore expression"
+
+let fix (getParseResultsForFile: GetParseResultsForFile) : CodeFix =
+  Run.ifDiagnosticByCode (set [ "20" ]) (fun diagnostic (codeActionParams: CodeActionParams) ->
+    asyncResult {
+      let fileName = codeActionParams.TextDocument.GetFilePath() |> Utils.normalizePath
+      let fcsPos = protocolPosToPos codeActionParams.Range.Start
+      let mDiag = protocolRangeToRange (UMX.untag fileName) diagnostic.Range
+
+      // Only do single line for now
+      if mDiag.StartLine <> mDiag.EndLine then
+        return []
+      else
+
+      let! (parseAndCheckResults: ParseAndCheckResults, _line: string, sourceText: IFSACSourceText) =
+        getParseResultsForFile fileName fcsPos
+
+      let mExprOpt =
+        (fcsPos, parseAndCheckResults.GetParseResults.ParseTree)
+        ||> ParsedInput.tryPick (fun path node ->
+          match node with
+          | SyntaxNode.SynExpr(e) when Range.equals mDiag e.Range -> Some(path, e)
+          | _ -> None)
+
+      match mExprOpt with
+      | None -> return []
+      | Some(path, expr) ->
+
+      let needsParentheses =
```

Comment:
@baronfel, the lack of indentation here is caused by [fsharp_experimental_keep_indent_in_branch](https://fsprojects.github.io/fantomas/docs/end-users/Configuration.html#fsharp_experimental_keep_indent_in_branch).

Would you be on board with this setting? I think it is useful for code fixes.

Reply:
Another strategy I've found useful to keep things flat is to factor out local functions at map/bind points, e.g.,

```fsharp
let fileName = codeActionParams.TextDocument.GetFilePath() |> normalizePath
let fcsPos = protocolPosToPos codeActionParams.Range.Start
let mDiag = protocolRangeToRange (UMX.untag fileName) diagnostic.Range

tryGetAst fileName fcsPos
|> AsyncResult.map (fun (ast, sourceText) ->
  tryGetExprInContext fcsPos mDiag ast
  |> Option.map (mkFix sourceText.GetLineString sourceText.GetSubTextFromRange)
  |> Option.toList)
```

or

```fsharp
asyncResult {
  let fileName = codeActionParams.TextDocument.GetFilePath() |> normalizePath
  let fcsPos = protocolPosToPos codeActionParams.Range.Start
  let mDiag = protocolRangeToRange (UMX.untag fileName) diagnostic.Range
  let! ast, sourceText = tryGetAst fileName fcsPos

  return
    tryGetExprInContext fcsPos mDiag ast
    |> Option.map (mkFix sourceText.GetLineString sourceText.GetSubTextFromRange)
    |> Option.toList
}
```

Altogether:

```fsharp
let fix (getParseResultsForFile: GetParseResultsForFile) : CodeFix =
  Run.ifDiagnosticByCode (Set.singleton "20") (fun diagnostic (codeActionParams: CodeActionParams) ->
    let tryGetAst fileName fcsPos =
      getParseResultsForFile fileName fcsPos
      |> AsyncResult.map (fun (parseAndCheckResults, _line, sourceText) -> parseAndCheckResults.GetAST, sourceText)

    let tryGetExprInContext fcsPos mDiag ast =
      (fcsPos, ast)
      ||> ParsedInput.tryPick (fun path node ->
        match node with
        | SyntaxNode.SynExpr e when Range.equals mDiag e.Range -> Some(path, e)
        | _ -> None)

    let mkFix getLineStr getSubtext (path, expr: SynExpr) =
      let needsParens =
        // expr |> ignore
        let exprInPipe =
          let lineNumber = expr.Range.StartLine

          let mkRangeInFile (offset: int) (length: int) : range =
            Range.mkRange
              expr.Range.FileName
              (Position.mkPos lineNumber (expr.Range.EndColumn + offset))
              (Position.mkPos lineNumber (expr.Range.EndColumn + offset + length))

          let mPipe = mkRangeInFile 2 2
          let mIgnore = mkRangeInFile 5 6

          SynExpr.App(
            ExprAtomicFlag.NonAtomic,
            false,
            SynExpr.App(
              ExprAtomicFlag.NonAtomic,
              true,
              SynExpr.LongIdent(
                false,
                SynLongIdent(
                  [ FSharp.Compiler.Syntax.Ident("op_PipeRight", mPipe) ],
                  [],
                  [ Some(IdentTrivia.OriginalNotation "|>") ]
                ),
                None,
                mPipe
              ),
              expr, // The expr that will now be piped into ignore.
              Range.unionRanges expr.Range mPipe
            ),
            SynExpr.Ident(FSharp.Compiler.Syntax.Ident("ignore", mIgnore)),
            Range.unionRanges expr.Range mIgnore
          )

        SynExpr.shouldBeParenthesizedInContext getLineStr path exprInPipe

      let newText =
        let currentText = getSubtext expr.Range

        if needsParens then
          $"(%s{currentText}) |> ignore"
        else
          $"%s{currentText} |> ignore"

      { SourceDiagnostic = None
        Title = title
        File = codeActionParams.TextDocument
        Edits =
          [| { Range = fcsRangeToLsp expr.Range
               NewText = newText } |]
        Kind = FixKind.Fix }

    asyncResult {
      let fileName = codeActionParams.TextDocument.GetFilePath() |> normalizePath
      let fcsPos = protocolPosToPos codeActionParams.Range.Start
      let mDiag = protocolRangeToRange (UMX.untag fileName) diagnostic.Range
      let! ast, sourceText = tryGetAst fileName fcsPos

      return
        tryGetExprInContext fcsPos mDiag ast
        |> Option.map (mkFix sourceText.GetLineString sourceText.GetSubTextFromRange)
        |> Option.toList
    })
```

But the indentation setting may also make sense.

Reply:
I'm less of a fan of this because when you read 

```fsharp
        tryGetExprInContext fcsPos mDiag ast
        |> Option.map (mkFix sourceText.GetLineString sourceText.GetSubTextFromRange)
        |> Option.toList
```

You need to look back up to see what `tryGetExprInContext` and `mkFix` do. 
And I think this is less ideal when debugging.